packages/retail-ui/components/Input/Input.tsx

```diff
@@ -402,7 +413,8 @@ class Input extends React.Component<InputProps, InputState> {
     });
 
     if (this.props.selectAllOnFocus) {
-      this.selectAll();
+      // https://github.com/facebook/react/issues/7769
+      this.delaySelectAll();
```

Comment:
```suggestion
      this.input ? this.selectAll() : this.delaySelectAll();
```
Я бы так сделал. Чтобы минимально менять поведение и тест не ломать.